Source/WebKit/Scripts/webkit/parser.py

```diff
@@ -180,3 +182,25 @@ def parse_parameters_string(parameters_string):
 
         parameters.append(model.Parameter(kind=parameter_kind, type=parameter_type, name=parameter_name, attributes=parse_attributes_string(attributes_string)))
     return parameters
+
+
+def parse_enabled_by_string(enabled_by_string):
+    enabled_by = None
+    enabled_by_conjunction = None
+
+    has_and_conjunction = '&&' in enabled_by_string
+    has_or_conjunction = '||' in enabled_by_string
+
+    if has_and_conjunction and has_or_conjunction:
+        raise Exception('ERROR: EnabledBy cannot contain both && and || conjunctions')
+    elif has_and_conjunction:
+        enabled_by = re.split(r'\s*&&\s*', enabled_by_string)
+        enabled_by_conjunction = '&&'
+    elif has_or_conjunction:
+        enabled_by = re.split(r'\s*\|\|\s*', enabled_by_string)
+        enabled_by_conjunction = '||'
+    else:
+        enabled_by = re.split(r'\s+', enabled_by_string)
```

Comment:
I think we should treat the whole string as a single flag in this case.
i.e. only allow && or || as a delimiter between flags.